ipam/mas.go

```diff
@@ -5,129 +5,193 @@ package ipam
-	queryInterval time.Duration
-	lastRefresh   time.Time
+	name       string
+	sink       addressConfigSink
+	fileLoaded bool
+	filePath   string
 }
 
 // MAS host agent JSON object format.
-type jsonObject struct {
-	Isolation string
-	IPs       []struct {
-		IP              string
-		IsolationId     string
-		Mask            string
-		DefaultGateways []string
-		DnsServers      []string
-	}
+type NetworkInterfaces struct {
+	Interfaces []Interface
+}
+
+type Interface struct {
+	MacAddress string
+	IsPrimary  bool
+	IPSubnets  []IPSubent
+}
+
+type IPSubent struct {
+	Prefix      string
+	IPAddresses []IPAddress
+}
+
+type IPAddress struct {
+	Address   string
+	IsPrimary bool
 }
 
 // Creates the MAS source.
 func newMasSource(options map[string]interface{}) (*masSource, error) {
-	queryUrl, _ := options[common.OptIpamQueryUrl].(string)
-	if queryUrl == "" {
-		queryUrl = masQueryUrl
-	}
-
-	i, _ := options[common.OptIpamQueryInterval].(int)
-	queryInterval := time.Duration(i) * time.Second
-	if queryInterval == 0 {
-		queryInterval = masQueryInterval
+	var filePath string
+	if runtime.GOOS == "windows" {
+		filePath = defaultWindowsFilePath
+	} else {
+		filePath = defaultLinuxFilePath
 	}
 
 	return &masSource{
-		name:          "MAS",
-		queryUrl:      queryUrl,
-		queryInterval: queryInterval,
+		name: "MAS",
+		filePath: filePath,
 	}, nil
 }
 
 // Starts the MAS source.
-func (s *masSource) start(sink addressConfigSink) error {
-	s.sink = sink
+func (source *masSource) start(sink addressConfigSink) error {
+	source.sink = sink
 	return nil
 }
 
 // Stops the MAS source.
-func (s *masSource) stop() {
-	s.sink = nil
+func (source *masSource) stop() {
+	source.sink = nil
 	return
 }
 
 // Refreshes configuration.
-func (s *masSource) refresh() error {
+func (source *masSource) refresh() error {
 
-	// Refresh only if enough time has passed since the last query.
-	if time.Since(s.lastRefresh) < s.queryInterval {
+	if source.fileLoaded {
 		return nil
 	}
-	s.lastRefresh = time.Now()
 
-	// Configure the local default address space.
-	local, err := s.sink.newAddressSpace(LocalDefaultAddressSpaceId, LocalScope)
+	// Query the list of local interfaces.
+	localInterfaces, err := net.Interfaces()
 	if err != nil {
 		return err
 	}
 
-	// Fetch configuration.
-	resp, err := http.Get(s.queryUrl)
+	// Query the list of Azure Network Interfaces
+	sdnInterfaces, err := getSDNInterfaces(source.filePath)
 	if err != nil {
 		return err
 	}
 
-	defer resp.Body.Close()
+	// Configure the local default address space.
+	local, err := source.sink.newAddressSpace(LocalDefaultAddressSpaceId, LocalScope)
+	if err != nil {
+		return err
+	}
+
+	err = populateAddressSpace(local, sdnInterfaces, localInterfaces)
+	if err != nil {
+		return err
+	}
 
-	// Decode JSON object.
-	var obj jsonObject
-	decoder := json.NewDecoder(resp.Body)
-	err = decoder.Decode(&obj)
+	// Set the local address space as active.
+	err = source.sink.setAddressSpace(local)
```

Comment:
we moved to this standard few months back. If the function returns only error, we can have this std. 

if err := source.sink.setAddressSpace(local); err != nil {
   return err
}